src/Contract/Operation/Unfoldable.php

```diff
@@ -6,19 +6,23 @@
 
 use loophp\collection\Contract\Collection;
 
+/**
+ * @template TKey
+ * @template T
+ */
 interface Unfoldable
 {
     /**
      * Create a collection by yielding from a callback with an initial value.
      *
      * @see https://loophp-collection.readthedocs.io/en/stable/pages/api.html#unfold
      *
-     * @template T
+     * @template U of T
```

Comment:
I'm not sure why this is needed - doesn't it work if we leave it like before, just `T` ? Either way I guess it's the same thing

Reply:
This is a static function, so, `U` must be defined locally at the method level. And `U` is going to be made of `T`.